### PR TITLE
Update schemas for parametric pulses

### PR DESCRIPTION
**Pull request by lcapelluto**
### Summary

Update schemas as necessary for #3463 

 - expects BackendConfiguration `parametric_pulses` field to be allowed
 - expects Qobj instruction with `name="parametric_pulse"` and `params` field

### CODE REVIEW

File: qiskit/schemas/qobj_schema.json [hunk 1]
```diff
@@ -152,6 +153,24 @@
...
                 },
+                {
+                    "properties": {
+                        "name": {
+                            "enum": [
+                                "parametric_pulse"
+                            ]
+                        },
+                        "pulse_shape": {"type": "string"},
+                        "params": {"type": "array"}
```
Comment (lcapelluto) [inline qiskit/schemas/qobj_schema.json]: actually a dict, how do I write that here? the keys are strings but the values aren't all the same type, and the keys aren't set either

File: qiskit/schemas/qobj_schema.json [hunk 2]
```diff
@@ -152,6 +153,24 @@
...
+                    },
+                    "required": [
+                        "pulse_shape",
+                        "params",
+                        "t0",
+                        "ch"
+                    ],
+                    "title": "Frame change pulse"
```
Comment (eggerdj) [inline qiskit/schemas/qobj_schema.json]: Shouldn't this be `"title": "Parametric Pulse"`?

Comment (lcapelluto) [reply qiskit/schemas/qobj_schema.json]: wow thank you yes 😱

File: qiskit/schemas/examples/qobj_openpulse_example.json [hunk 3]
```diff
@@ -33,7 +33,8 @@
             {"name": "acquire", "t0": 10, "duration": 90, "qubits": [0,1,2], "register_slot": [0,1,2], "memory_slot": [0,1,2]},
             {"name": "copy", "register_orig": 1, "register_copy": [3,4]},
             {"name": "bfunc", "mask": "0xE", "relation": "==", "val": "0xA", "register": 3, "memory": 3},
-            {"name": "pulse1", "ch": "d0", "t0": 120, "conditional": 3}
+            {"name": "pulse1", "ch": "d0", "t0": 120, "conditional": 3},
+            {"name": "parametric_pulse", "t0": 0, "params": {"amp": [0.0, 0.5], "sigma": 4, "duration": 25}, "ch": "d0", "pulse_shape": "gaussian"},
```
Comment (diego-plan9) [inline qiskit/schemas/examples/qobj_openpulse_example.json]:
```suggestion
            {"name": "parametric_pulse", "t0": 0, "params": {"amp": [0.0, 0.5], "sigma": 4, "duration": 25}, "ch": "d0", "pulse_shape": "gaussian"}
```

Small suggestion for the `json.decoder.JSONDecodeError: Expecting value: line 38 column 13 (char 1594)` in current travis run - json is a bit picky when it comes to commas after the last element of an array.

File: qiskit/schemas/qobj_schema.json [hunk 4]
```diff
@@ -152,6 +153,24 @@
...
                 },
+                {
+                    "properties": {
+                        "name": {
+                            "enum": [
+                                "parametric_pulse"
+                            ]
+                        },
+                        "pulse_shape": {"type": "string"},
+                        "params": {"type": "object"}
```
Comment (diego-plan9) [inline qiskit/schemas/qobj_schema.json]: This is the conflicting `params` from the main comment (along with the more general at line 262).

**Comment by lcapelluto** [comment]
Thank you, @diego-plan9 :) I think the best option would be changing the name. The type is different, so we should try to treat them as different things. Unfortunately, the only suggestion I have is `parameters`, which is what they are called by #3620.

**Comment by taalexander** [comment]
If params were an array (like it is for the QASM path) this wouldn't be an issue. This would increase consistency between the two paths.

**Comment by lcapelluto** [comment]
@taalexander perhaps we should change the other `params` to an `object`

**Comment by taalexander** [comment]
I would rather not, the other params is the much more heavily used qasm path. I still prefer the list as an argument as it is closer to an assembly input.

**Pull request merged**